src/MiniGrafx.cpp

```diff
@@ -471,7 +471,7 @@ uint16_t MiniGrafx::getStringWidth(const char* text, uint16_t length) {
   uint16_t maxWidth = 0;
 
   while (length--) {
-    stringWidth += readFontData(fontData, JUMPTABLE_START + (text[length] - firstChar) * JUMPTABLE_BYTES + JUMPTABLE_WIDTH);
+    stringWidth += readFontData(fontData, JUMPTABLE_START + ((reinterpret_cast<const unsigned char*>(text)[length]) - firstChar) * JUMPTABLE_BYTES + JUMPTABLE_WIDTH);
```

Comment:
Any idea how well supported this is across the platforms supported by this library?

Reply:
I am not sure if I understand your question correct.
Do you mean if signed chars are usually used on platforms that are supported by the library?
I cannot tell you. In fact, I found this issue because I am writing a simple emulation environment for the PC. That is, I write my sketch on the PC with its better debugging possibilities, before using it on an esp8266. I got these crashes, then I located the issue. I did a cross-check with -funsigned-char, so I could confirm that the signedness is the problem, so I came up with this patch.

Having said this: Even if no supported platform used signed chars, the paradigm of defensive programming would tell us that it is not good to rely on the unsignednes of char.

Reply:
> Do you mean if signed chars are usually used on platforms that are supported by the library?

Sorry, I meant the `reinterpret_cast` bit which is new to me.

> ...emulation environment for the PC. That is, I write my sketch on the PC with its better debugging possibilities

That sounds great!

> defensive programming would tell us that it is not good to rely on the unsignednes of char

Here it gets a bit philosophical and I disagree. I am all for defensive programming but _within the bounds of the supported platforms_. If a component is specified to work on platforms A & B then that's likely the platforms its authors test the component against. If the component happens to also work on platform C then that's great - if it doesn't it's by means a surprise. It will take someone like you with enough interest in support for platform C to fix this 😄

Reply:
The reinterpret_cast is there from the first version of ANSI C++. My "Stroustrup, die C++ Programmiersprache, 2nd edition" from 1992 already contains it, and the book is pre-ANSI/ISO. Of course, if you do not like this, this could be changed to a "normal", that is, C-style cast. That's not a problem.
Another (and, most likely, much cleaner) option might be to change the definition of fontData to be of type unsigned char. As this changes the public interface, I am not sure if this does not create more problems, though, especially in dependant projects.

Regarding my emulation environment, it is very limited at the moment, implementing only what is needed for my project. As it also has an option to scale the graphical output, it was the reason why I found the other bug with the missing lower-right corner (#58). On the real display, it was too small to see it, at least for me.

Regarding defensive programming: What I meant, IF I know something can be problematic, I try to make sure it does not happen. For example, the signed/unsigned char problematic has bit me so often in the past that I am sensitive to this in my own code. Of course, if one is not aware of a problem, it is unlikely one will handle it defensive in the first place. In this case, it needs someone to find out that there is a problem, and this often happens "the hard way".

What I wanted to say with this comment: As the problem has been identified now, I hope you can add the fix as a matter of making the code more defensive.

Just to make sure: This comment was *not* meant as a complaint against your project!

Reply:
> This comment was _not_ meant as a complaint against your project!

Oh, no worries, I didn't take it as such. As I didn't write any of the code here it's easy for me to maintain a "healthy" distance.

Thanks for the fixes.

Reply:
Since most of this library relies on just C functionality i would suggest changing this to:
```diff
-    stringWidth += readFontData(fontData, JUMPTABLE_START + ((reinterpret_cast<const unsigned char*>(text)[length]) - firstChar) * JUMPTABLE_BYTES + JUMPTABLE_WIDTH);
+    stringWidth += readFontData(fontData, JUMPTABLE_START + (((const unsigned char *)(text)[length]) - firstChar) * JUMPTABLE_BYTES + JUMPTABLE_WIDTH);
```